src/__test__/components/data-exploration/differential-expression-tool/DiffExprCompute.test.jsx

```diff
@@ -1,244 +1,286 @@
 
-const initialState = {
-  cellSets: {
-    properties: {
-      'cluster-a': {
-        name: 'cluster a',
-        key: 'cluster-a',
-        cellIds: new Set([1, 2]),
-        color: '#00FF00',
-      },
-      'cluster-b': {
-        name: 'cluster b',
-        key: 'cluster-b',
-        cellIds: new Set([3, 4, 5]),
-        color: '#FF0000',
-      },
-      'cluster-c': {
-        name: 'cluster c',
-        key: 'cluster-c',
-        cellIds: new Set([6]),
-        color: '#0000FF',
-      },
-      'sample-a': {
-        name: 'sample a',
-        key: 'sample-a',
-        cellIds: new Set([1, 2]),
-        color: '#00FF00',
-      },
-      'sample-b': {
-        name: 'sample b',
-        key: 'sample-b',
-        cellIds: new Set([3, 4, 5]),
-        color: '#FF0000',
-      },
-      'sample-c': {
-        name: 'sample c',
-        key: 'sample-c',
-        cellIds: new Set([6]),
-        color: '#0000FF',
-      },
-      louvain: {
-        name: 'Louvain clusters',
-        key: 'louvain',
-        type: 'cellSets',
-        cellIds: new Set(),
-        rootNode: true,
-      },
-      scratchpad: {
-        name: 'Custom selections',
-        key: 'scratchpad',
-        type: 'cellSets',
-        cellIds: new Set(),
-        rootNode: true,
-      },
-      sample: {
-        name: 'Samples',
-        key: 'sample',
-        type: 'metadataCategorical',
-        cellIds: new Set(),
-        rootNode: true,
-      },
-    },
-    hierarchy: [
-      {
-        key: 'louvain',
-        children: [{ key: 'cluster-a' }, { key: 'cluster-b' }, { key: 'cluster-c' }],
-      },
-      {
-        key: 'sample',
-        children: [{ key: 'sample-a' }, { key: 'sample-b' }, { key: 'sample-c' }],
-      },
-      {
-        key: 'scratchpad',
-        children: [],
-      },
-    ],
-  },
-  differentialExpression: {
-    properties: {
-      data: [],
-      comparisonGroup: {},
-      loading: false,
-      error: false,
-      total: 0,
-    },
-    comparison: {
-      type: 'between',
-      group: {
-        between: {
-          cellSet: null,
-          compareWith: null,
-          basis: null,
-        },
-      },
-    },
-  },
+import mockAPI, { generateDefaultMockAPIResponses } from '__test__/test-utils/mockAPI';
+import fake from '__test__/test-utils/constants';
+
+enableFetchMocks();
+
+const createMockStore = configureMockStore([thunk]);
+
+const mockOnCompute = jest.fn();
+const experimentId = fake.EXPERIMENT_ID;
+
+const defaultProps = {
+  experimentId,
+  onCompute: mockOnCompute,
 };
+const DiffExprComputeFactory = createTestComponentFactory(DiffExprCompute, defaultProps);
 
-describe('DiffExprCompute', () => {
-  it('renders correctly with no comparison method', () => {
-    const store = mockStore(initialState);
+const mockAPIresponses = generateDefaultMockAPIResponses(experimentId);
 
-    const component = mount(
+const renderDiffExprCompute = async (store) => {
+  await act(() => {
```

Comment:
my editor says `await has no effect on the type of this expression`. Are you sure we need await here?

Reply:
Oh it's because the callback isn't added with `async`. We should be adding an `async` here because we want to wait for some funtion calls to wait.

Reply:
A referece about this : https://medium.com/@bmb21/reacts-sync-and-async-act-caa297b658b0